### PR TITLE
Slimepeople and slimes are now allergic to salt

### PR DESCRIPTION
**Pull request by cebutris**
:cl:  Cebutris
add: Slimes and Slimepeople are now hurt by salt! Be careful!
/:cl:

Have you ever seen what happens when you put salt on snails? It's terrifying and interesting. 

Slimes are now snails, as far as salt is concerned

### CODE REVIEW

**Comment by AlexTheSysop42** [comment]
does this extend to slimes?

**Comment by Kraseo** [comment]
but why

**Comment by cebutris** [comment]
It extends to every variant of slimeperson, but not the mob slimes

**Comment by Kraseo** [comment]
also these are the other kind of slime people

**Comment by AlexTheSysop42** [comment]
mob slimes or riot

**Comment by AlexTheSysop42** [comment]
no all slimepeople are subtypes of jelly now

**Comment by cebutris** [comment]
@Kraseo all slimepeople are a child of jellies

**Comment by cebutris** [comment]
It now works with mob slimes, with a slight bit of snowflakecoding

**Comment by Kromgar** [comment]
inb4 salty xenobio mains appear

**Comment by cebutris** [comment]
it purges at a rate of 10u per cycle, so 10u will only trigger the damage once

**Comment by 81Denton** [comment]
I was gonna say "this will be OP in spray bottles" but then remembered that you can do much worse with phlog+beepsky smash+facid etc.

**Comment by CameronWoof** [comment]
Consider: anti-revenant smoke machining is now anti-slimeperson smoke machining. Is there a balance reason for this change?

**Comment by 81Denton** [comment]
You can just put on a firesuit and turn it off mate

**Comment by CameronWoof** [comment]
Okay, but slimepeople already have a bunch of snowflake medbay stuff they need to deal with. Why do they need another downside?

**Comment by 81Denton** [comment]
>another downside

If docs are incompetent and inject you with charcoal, that's an IC issue.

**Comment by CameronWoof** [comment]
The fact that it's an IC issue that incompetent doctors can malpractice you to death with charcoal or cryo does not make it less of a downside for the race.

**Comment by 81Denton** [comment]
That's pretty trivial compared to the huge advantages like 50% reduced burn damage, TRAIT_TOXINLOVER, body splitting and limb regeneration.

**Comment by CameronWoof** [comment]
50% reduced burn damage is already paid for by the (if I'm not mistaken) larger weakness to cold damage, and toxinlover will be to your advantage less often than than getting hit with charcoal/tricord/a-tox/etc. will be to your disadvantage.

That said, body splitting and limb regeneration are why slimes aren't a roundstart race (in downstreams where they are, they tend not to have those features.) They're a xenobio race, they *should* bring something else to the table beyond two advantages that are more than paid for by strict disadvantages.

**Comment by ExcessiveUseOfCobblestone** [comment]
Not everything has to be net objectively better or net objectively worse. It's a (largely) opt-in feature.

I just think this is funny because I call dwarfy a slug.

**Comment by cebutris** [comment]
I didn't consider any balance implications when I made this, I just thought it would be funny. I don't think it's a big deal as far as balance is concerned, though. It's not like salt is any more common than antitoxin which is worse for slimes than salt.

**Comment by AlexTheSysop42** [comment]
this would be a pain if a slimeperson gets a virus that has salt as a cure

**Comment by ExcessiveUseOfCobblestone** [comment]
You'd just have to ask for a pinch of salt then @AlexTheSysop42

File: code/modules/mob/living/carbon/human/species_types/jellypeople.dm [hunk 1]
```diff
@@ -15,6 +15,14 @@
 	heatmod = 0.5 // = 1/4x heat damage
 	burnmod = 0.5 // = 1/2x generic burn damage
 
+/datum/species/jelly/handle_chemicals(datum/reagent/chem, mob/living/carbon/human/H)
+	if(chem.id == "sodiumchloride")
+		H.adjustFireLoss(5)
```
Comment (lordpidey) [inline code/modules/mob/living/carbon/human/species_types/jellypeople.dm]: This is silly, 0.1 will deal the exact same amount of damage as 10.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 2]
```diff
@@ -345,6 +345,9 @@
 	if(reac_volume < 1)
 		return
 	new/obj/effect/decal/cleanable/salt(T)
+	for(var/mob/living/simple_animal/slime/M in T)
+		M.adjustBruteLoss(reac_volume)
+		M.visible_message("<span class='warning'>[M] bubbles and froths, leaking liquid everywhere!","<span class='userdanger'>You feel an agonizing burning as you melt away!")
```
Comment (duncathan) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: why not put this is `reaction_mob()`?

**Comment by YakumoChen** [comment]
@AlexTheSysop42 good thing slimepeople are already immune to diseases then!

**Comment by AlexTheSysop42** [comment]
I thought xdtm removed that so they wouldn't be immune to advanced mutation toxin

**Comment by YakumoChen** [comment]
i dont think reagents are the same thing as diseases?

**Comment by AlexTheSysop42** [comment]
But advanced mutation toxin worked by giving them a disease

**Comment by AlexTheSysop42** [comment]
and the disease was supposed to instantly reach stage 5 on slimepeople

**Comment by duncathan** [comment]
@cebutris

**Comment by fludd12** [comment]
@AlexTheSysop actually the reagent doesn't apply the disease if they're a slimeperson, IIRC. Just morphs them instantly.

File: code/modules/mob/living/carbon/human/species_types/jellypeople.dm (hunk 1 above)
Comment (fludd12) [reply code/modules/mob/living/carbon/human/species_types/jellypeople.dm]: Wouldn't that issue apply to almost any toxin that you can add an amount less than its absorption rate, though?

File: code/modules/reagents/chemistry/reagents/food_reagents.dm (hunk 2 above)
Comment (cebutris) [reply code/modules/reagents/chemistry/reagents/food_reagents.dm]: This is the same code that cryostylane uses. How would I make `reaction_mob` only apply to slimes?

**Comment by KorPhaeron** [comment]
Ping for reopen

**Pull request closed without merging**